### PR TITLE
Add fallback-recommendations-skill

### PR DESCRIPTION
**Pull request by LinusSkucas**
## Info

This PR adds the new skill, [fallback-recommendations-skill](https://github.com/LinusS1/fallback-recommendations-skill), to the skills repo.

## Description

When Mycroft doesn't know something, he will look it up skills that could help you complete your intent. If he finds a skill, he will download it to your device.

<sub>Created with [mycroft-skills-kit](https://github.com/mycroftai/mycroft-skills-kit) v0.3.12</sub>

### CODE REVIEW

**Comment by LinusSkucas** [comment]
Testing instructions:
https://github.com/LinusS1/fallback-recommendations-skill/blob/master/TESTING.md

**Comment by KathyReid** [comment]
## Meta

* Platform: Ubuntu Linux 18.04 LTS
* Device version: 18.8.9
* Who: @KathyReid / @kathy-mycroft
* Datestamp: Fri 28 Dec 11:08:16 AEDT 2018
* Language and dialect of tester: English native speaker, Australian accent

## Standard checks

* Check that the Skill installs using voice commands. Mycroft will get the user to confirm which Skill should be installed if there is ambiguity in Skill names - such as duplicate names. If possible, name the Skill so that there is minimal duplication and/or conflict. You should also verify that the Skill name can be verbally pronounced by speaking the Skill name into the Mycroft command line several times, and reading the resulting transcriptions. Suggest alternative Skill names if it is difficult to verbally pronounce the Skill name.

> this is the first time the Skill has been submitted to the `mycroft-skills` repo so it won't yet be available for voice installation. I will check the STT utterance hears the Skill name correctly. 

```
~~~~264 - __main__:handle_utterance:61 - INFO - Utterance: ["it's to full-back recommendations"]
```

```
~~~~962 - __main__:handle_utterance:61 - INFO - Utterance: ['install full-back recommendations']
```

> It's having a bit of trouble with transcribing the name of the Skill correctly, but I also have hayfever at the moment, so that's probably part of the poor transcription too. Falling back to `msm` installation. 

```
kathyreid@kathyreid-ubuntu-X510UQ:~/mycroft-core$ source venv-activate.sh 
Entering Mycroft virtual environment.  Run 'mycroft-venv-deactivate' to exit
(.venv) kathyreid@kathyreid-ubuntu-X510UQ:~/mycroft-core$ msm install https://github.com/LinusS1/fallback-recommendations-skill
WARNING - No such default skill: plasma-user-control-skill
INFO - Downloading skill: https://github.com/LinusS1/fallback-recommendations-skill
INFO - Installing requirements.txt for fallback-recommendations-skill
INFO - Successfully installed fallback-recommendations-skill
```
> installed successfully

* Check the `dialog` directory to ensure that from a voice user interface perspective the dialogs read well. It is a good idea to run the `dialog` phrases through [mimic](https://mycroft.ai/documentation/mimic/). Sometimes the `dialog` files will need a small tweak such as a space between words, or extra vowel sounds, to sound realistic. Sometimes words don't render as expected and alternative wording should be used. Alway play every `dialog` phrase on the command line by doing `say <your dialog phrase here>` so that you can check how the `dialog` is spoken. Some of the tricks you might need to use include separating words by a space - such as `sub sonic` instead of `subsonic`, or `broad cast` instead of `broadcast`.

> ran each of the `dialog` files through `mimic`. Pronunciations are good. 

* Check the **Intents** defined in the `vocab` diectory. Determine if the **Intents** are likely to conflict with those from **Skills** that are installed by default, and if so, suggest alternatives if this makes sense, or at least advise the Skill Author that there are likely to be `Intent Collisions`. [Integration Test Runner](https://mycroft.ai/documentation/skills/automatic-testing/) can also assist with this.

> This is a _fallback_ Skill, there are no `vocab` files used, skipping check

_NOTE: Skills that are installed by default vary by `enclosure` and can be found in the `mycroft-skills` repo in the files called `DEFAULT.enclosure`. For example, the default Skills for Picroft are in a file called `DEFAULT.picroft`._

* Check the regular expressions defined in the `regex` directory - do they appear sensible? [Integration Test Runner](https://mycroft.ai/documentation/skills/automatic-testing/) can also assist with this.

> `regex` not used, skipping check

* Check README for completeness - does it follow the README template and include all the relevant sections such as Intents, known issues, dependencies and so on? Is at least one category selected? _At least one category must be selected for the Skill to be displayed correctly in the [Mycroft Marketplace](https://market.mycroft.ai)._

> There is a broken image link in the README.md file, otherwise it reads well. 

_ACTION REQUIRED: Please pull in [this PR](https://github.com/LinusS1/fallback-recommendations-skill/pull/1) to fix the broken image_

* Check `requirements.txt` and `requirements.sh` - are the required dependencies listed? If `requirements.sh` is used, is some form of conditional processing done to match against multiple distros? Often Skill Authors will add requirements.txt using only an “library=1.x.x” instead of “library >=1.x.x”. Check to make sure that here is an equal or greater than in the requirements to help future-proof the Skill, unless a _specific_ version is needed. 

> There is only one requirement, no version specified, all OK

* Check `settingsmeta.json` - are the settings well laid out? Does the placeholder text make sense?

> `settingsmeta.json` not used, skipping check 

## Python code checks

* Is an appropriate LICENSE file used in the repo - such as Apache2 or GPL? 
> Yes, Apache license

* Inline documentation - is the code documented in line?
> There is not a lot of inline documentation, however it's a fairly straightforward Skill

* Libraries - does the Skill include the correct libraries? Does it use too many libraries or dependencies?
> Just one library used, no version specified

* Callback check - do we need to do a check for the callback error that was in the Spotify Skill?
> n/a

* Error handling - are there any specific checks we make for error handling or graceful degradation?
> error handling is done well

* Automatic tests - do automatic tests of the **Skill** pass?
> yes

* Skill Settings using Python Booleans - Skill Settings were updated in August 2018 to use Python Booleans instead of text Boolean values used in Python 2.4. [This is a good example of the change to check for](https://github.com/MycroftAI/skill-date-time/pull/32/files?utf8=%E2%9C%93&diff=split)
> not required 

* As this is a Fallback Skill, I am checking for the order of precedence, many thanks to @penrods for the below information: 

```
8, Persona (https://github.com/MycroftAI/skill-fallback-persona/blob/master/__init__.py#L27)
10, Wolfram Alpha (https://github.com/MycroftAI/skill-fallback-persona/blob/master/__init__.py#L27),
10, DuckDuckGo (https://github.com/MycroftAI/fallback-duckduckgo/blob/18.08/compatibility.py#L67)
100, Unknown (https://github.com/MycroftAI/fallback-unknown/blob/18.08/__init__.py#L25)
```
This Fallback Skill has an order of precedence of 17, which we think is "about right". 

_NOTE: We do not enforce PEP8 Checks on Skills_

## Functional checks 

Thank you for including information on how to test, appreciate it :+1: 

* `Hey Mycroft, how do I mix a margarita?`
> Correctly downloaded and installed the Cocktails Skill
> Correctly triggered the intent from the Cocktails Skill

* `Hey Mycroft, give me a rat fact`
> This downloaded and installed the Wikiquote Skill, which I wasn't expecting
> The Wikiquote Skill failed to fulfil the intent
(but I still think the Skill's actions were correct, I'm counting this as a pass)

_removed Wikiquote skill_

* `Hey Mycroft, give me a quote by Radia Perlman`
> Correctly downloaded and installed the Wikiquote Skill
> It didn't give me a quote by Radia but it may not have any. It **did** give me a quote
> Skill worked as expected

#### Test case that should not match a Skill

* `Hey Mycroft, blargh argh blerp`
> Intent failure

* `Hey Mycroft, swish wish bish`
> It downloaded OpenHAB (!) - there must be a matching intent in that Skill

_In summary this is functionally working really well_

## Summary

This is a really well-written Skill, thank you so much. There is just one minor thing to fix before we can merge it in - the icon in the README is broken; I've raised a PR which fixes it. 

* Please fix the icon in the README - there is a PR on your repo for this

**Comment by KathyReid** [comment]
Hey @LinusS1 I found another minor issue after using this for a while.
If the Fallback installs a Skill once, it does not subsequently check in that Skill is already installed.

```
21:06:24.741 - Skill Recommendations Fallback - INFO - lifx-mycroft
21:06:27.925 - msm.mycroft_skills_manager - INFO - Best match (1.0): lifx-mycroft by samclane
~~~~05 - msm.mycroft_skills_manager - WARNING - No such default skill: plasma-user-control-skill
~~~~01 - msm.mycroft_skills_manager - WARNING - No such default skill: plasma-user-control-skill
21:06:36.003 - mycroft.skills.core:handler:1508 - ERROR - Exception in fallback.
Traceback (most recent call last):
  File "/home/kathyreid/mycroft-core/mycroft/skills/core.py", line 1499, in handler
    if handler(message):
~~~~oft/skills/fallback-recommendations-skill.linuss1/__init__.py", line 123, in handle_fallback
    self.install_skill(skill)
~~~~ycroft/skills/fallback-recommendations-skill.linuss1/__init__.py", line 96, in install_skill
    skill.install()
~~~~eid/mycroft-core/.venv/lib/python3.6/site-packages/msm/skill_entry.py", line 238, in install
    raise AlreadyInstalled(self.name)
msm.exceptions.AlreadyInstalled: lifx-mycroft
  ^--- NEWEST ---^ 

```

**Comment by KathyReid** [comment]
Hi @LinusS1 let me know if you have any questions - this one should be reasonably easy to finalise to get into the `mycroft-skills` repo :+1:

**Comment by LinusSkucas** [comment]
Okay, sorry for the delay I'm on a trip but once I get back, I'll be able to push it to the skills repo.

**Comment by LinusSkucas** [comment]
alright, fixed everything, new pr: #739. I'm not sure why the travis checks are failing, but they used to be good.
